test/Microsoft.AspNet.Mvc.Core.Test/ModelBindingHelperTest.cs

```diff
@@ -17,15 +17,29 @@ namespace Microsoft.AspNet.Mvc.Test
 {
     public class ModelBindingHelperTest
     {
-        [Fact]
-        public async Task TryUpdateModel_ReturnsFalse_IfBinderReturnsNull()
+        public static TheoryData<ModelBindingResult> UnsuccessfulModelBindingData
+        {
+            get
+            {
+                return new TheoryData<ModelBindingResult>
+                {
+                    null,
+                    new ModelBindingResult("someKey"), // IsFatalError true as well as IsModelSet false.
+                    new ModelBindingResult(model: null, key: "someKey", isModelSet: false),
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(UnsuccessfulModelBindingData))]
+        public async Task TryUpdateModel_ReturnsFalse_IfBinderIsUnsuccessful(ModelBindingResult binderResult)
         {
             // Arrange
             var metadataProvider = new EmptyModelMetadataProvider();
-
             var binder = new Mock<IModelBinder>();
-            binder.Setup(b => b.BindModelAsync(It.IsAny<ModelBindingContext>()))
-                  .Returns(Task.FromResult<ModelBindingResult>(null));
+            binder
+                .Setup(b => b.BindModelAsync(It.IsAny<ModelBindingContext>()))
+                .Returns(Task.FromResult<ModelBindingResult>(binderResult));
```

Comment:
The mock doesn't have to `throw`. Could just `Verify()` that `Validate()` is called `Times.Zero`.